### DISCUSSION TITLE
Last Motion

### DISCUSSION THREAD

**Discussion by Sblackk** [indomus/forum#108] [HUB: Home Assistant]
Salve...!!! Sono alle prese, per la prima volta, con Home Assistant (distribuzione HASSIO) installato su di un raspberry da circa una settimana. Ho effettuato tutte le automazioni che mi interessavano ma... vorrei una "scorciatoia" per aggirare questo ostacolo. Dovrei creare una scheda Lovelace, e quindi un'automazione, dove mi vengano indicati gli ultimi 5 movimenti rilevati da 2 sensori. Ho creato una variabile "ultimo_movimento" che non riesco a far valorizzare. Di seguito il blocco. Grazie in anticipo a chi mi dedica qualche minuto.

```- id: '1616956649217'
  alias: Aggiornamento Ultimo Movimento-
  description: Ultimo Movimento
  trigger:
  - platform: state
    to: 'on'
    entity_id: binary_sensor.mode_motion
  - platform: state
    entity_id: binary_sensor.cam_salotto_motion_trigger
    to: 'on'
  condition: []
  action:
  - service: variable.set_variable
    data:
      variable: ultimo_movimento
      attributes_template:
        {
           "history_1": "{{ variable.state }}",
           "history_2": "{{ variable.attributes.history_1 }}",
           "history_3": "{{ variable.attributes.history_2 }}",
           "history_4": "{{ variable.attributes.history_3 }}",
           "history_5": "{{ variable.attributes.history_4 }}",
          }
    data_template:
      value: >
        {{ trigger.to_state.attributes.friendly_name }}:
        {{ as_timestamp(states.variable.ultimo_movimento.last_changed)| timestamp_custom('%X') }}
  initial_state: 'on'
  mode: single```

**Comment by snakuzzo** [comment]
Interessante...però sinceramente eviterei di sfruttare quel custom component per le variabili.
Proverei a farla sfruttando funzionalità native. Si potrebbe sfruttare un JSON.,,ci devo pensare.

Sicuramente si può fare tramite uno script Python (non un `Python Script` di HA) in modo molto semplice memorizzando un file JSON e poi creando un sensore che legga quei valori...più ovviamente l'automazione per scrivere il nuovo valore quando viene rilevato.
